kong/client.go

```diff
@@ -164,6 +165,34 @@ func NewClient(baseURL *string, client *http.Client) (*Client, error) {
 	return kong, nil
 }
 
+// WithWorkspace return the client with a defined workspace
+func (c *Client) WithWorkspace(workspace string) *Client {
+	c.SetWorkspace(workspace)
+	return c
+}
+
+// SetWorkspace defines a workspace for the client
+func (c *Client) SetWorkspace(workspace string) {
+	c.workspace = workspace
+}
+
+//ClearWorkspace clear the workspace
+func (c *Client) ClearWorkspace() {
+	c.workspace = ""
```

Comment:
This needs to be synchronized, so it might be best to use `SetWorkspace`:
```suggestion
	c.SetWorkspace("")
```

Or maybe we don't need this function at all, and `SetWorkspace("")` should be the public interface to clear workspace? @hbagdi @mmorel-35 do you see use for `ClearWorkspace` as opposed to `SetWorkspace("")`?